### PR TITLE
Mwalker/lnn

### PR DESCRIPTION
**Pull request by killyosaur**
So you guys can start looking at what my changes are and see if I am nuts or not :P

### CODE REVIEW

**Review by anarasimham — APPROVED** [review]
Overall looks good.

I wouldn't generally put config parameters into anything but the top level file (run.py) but given this is a research project and we have a week left it doesn't matter too much.

LNN looks intense! Didn't get into that too much, will leave those details to you.

I haven't tested it to make sure it runs but assuming it does, good to go.

The refactoring makes sense - splitting functions into relevant files would be good when we scale this thing out.

File: .vscode/launch.json
Comment (sfwatanabe) [inline .vscode/launch.json:1]: Not sure we need to check this one in, but I'm indifferent either way.

**Review by sfwatanabe — APPROVED** [review]

File: .vscode/launch.json
Comment (killyosaur) [reply .vscode/launch.json:1]: I just missed it. will remove on future changes

**Comment by killyosaur** [comment]
> Overall looks good.
> 
> I wouldn't generally put config parameters into anything but the top level file (run.py) but given this is a research project and we have a week left it doesn't matter too much.
> 
> LNN looks intense! Didn't get into that too much, will leave those details to you.
> 
> I haven't tested it to make sure it runs but assuming it does, good to go.
> 
> The refactoring makes sense - splitting functions into relevant files would be good when we scale this thing out.

Gonna just say this: never look at Kubernetes configs then, they will make you lose your mind :D

having worked on production level configs, not putting all config elements at the root is pretty standard practice as it allows us to group up sub configs more effectively. But to each their own :)

**Comment by anarasimham** [comment]
Did some searching around and it looks like you can do it either way - learned something new

**Pull request merged**